roachtest: sqlsmith/setup=empty/setting=no-mutations failed

roachtest.sqlsmith/setup=empty/setting=no-mutations [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3012796&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3012796&tab=artifacts#/sqlsmith/setup=empty/setting=no-mutations) on release-21.1 @ [b600d9c2283e19886406e658369e70b5c61a612f](https://github.com/cockroachdb/cockroach/commits/b600d9c2283e19886406e658369e70b5c61a612f):


```
The test failed on branch=release-21.1, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=empty/setting=no-mutations/run_1
	sqlsmith.go:206,sqlsmith.go:237,test_runner.go:767: error: pq: internal error: st_node(): unknown LineMerge result type: *geom.GeometryCollection
		stmt:
		WITH
			with_2736 (col_15046)
				AS (
					SELECT
						*
					FROM
						(
							VALUES
								(
									st_node(st_linefromencodedpolyline(e'&\x11LdqK\x03':::STRING::STRING)::GEOMETRY::GEOMETRY)::GEOMETRY
								),
								(NULL)
						)
							AS tab_8687 (col_15046)
				)
		SELECT
			cte_ref_837.col_15046 AS col_15047
		FROM
			with_2736 AS cte_ref_837
		LIMIT
			65:::INT8;
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh sqlsmith/setup=empty/setting=no-mutations
```
</p>
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=empty/setting=no-mutations.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/geo/geomfn/node.go
+++ b/pkg/geo/geomfn/node.go
@@ -65,6 +65,12 @@ func Node(g geo.Geometry) (geo.Geometry, error) {
 	switch t := glines.(type) {
 	case *geom.MultiLineString:
 		mllines = t
+	case *geom.GeometryCollection:
+		if t.Empty() {
+			t.SetSRID(int(g.SRID()))
+			return geo.MakeGeometryFromGeomT(t)
+		}
+		return geo.Geometry{}, errors.AssertionFailedf("unknown GEOMETRYCOLLECTION: %T", t)
 	default:
 		return geo.Geometry{}, errors.AssertionFailedf("unknown LineMerge result type: %T", t)
 	}

--- a/pkg/geo/geomfn/node_test.go
+++ b/pkg/geo/geomfn/node_test.go
@@ -31,6 +31,12 @@ func TestNode(t *testing.T) {
 			false,
 		},
 		{
+			"LineString, same point",
+			geo.MustParseGeometry("SRID=4326;LINESTRING (-0.06435 -0.06948, -0.06435 -0.06948)"),
+			geo.MustParseGeometry("SRID=4326;GEOMETRYCOLLECTION EMPTY"),
+			false,
+		},
+		{
 			"LineString, 4 nodes",
 			geo.MustParseGeometry("LINESTRING(0 0, 10 10, 0 10, 10 0, 10 10)"),
 			geo.MustParseGeometry("MULTILINESTRING((0 0,5 5),(5 5,10 10),(10 10,0 10,5 5),(5 5,10 0,10 10))"),

